StreamChatClient_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -9,378 +9,257 @@ class WebSocketClient {
+  private var activeBackgroundTask: UIBackgroundTaskIdentifier?
+
+  /// An object containing external dependencies of `WebSocketClient`
+  private let environment: Environment
 
   init(
     urlRequest: URLRequest,
     eventDecoder: AnyEventDecoder,
-    callbackQueue: DispatchQueue,
+    eventMiddlewares: [EventMiddleware],
+    reconnectionStrategy: WebSocketClientReconnectionStrategy = DefaultReconnectionStrategy(),
     environment: Environment = .init()
   ) {
     self.environment = environment
-    self.engine = environment.engineBuilder(urlRequest, callbackQueue)
-    self.notificationCenter = environment.notificationCenter
+    self.urlRequest = urlRequest
+    self.middlewares = eventMiddlewares
+    self.reconnectionStrategy = reconnectionStrategy
     self.eventDecoder = eventDecoder
-    engine.delegate = self
-  }
-}
-
-extension WebSocketClient {
-  /// Connect to web socket.
-  /// - Note:
-  ///     - Skip if the Internet is not available.
-  ///     - Skip if it's already connected.
-  ///     - Skip if it's reconnecting.
-  public func connect() {
-    cancelBackgroundWork()
-
-//        if isConnected || connectionState == .connecting || connectionState == .reconnecting {
-//            if let logger = logger {
-//                let reasons = [(isConnected ? " isConnected with connectionId = \(connectionId ?? "n/a")" : nil),
-//                               (connectionState == .reconnecting ? " isReconnecting" : nil),
-//                               (connectionState == .connecting ? "isConnecting" : nil),
-//                               (provider.isConnected ? "\(provider).isConnected" : nil)]
-//
-//                logger.log("SKIP connect: \(reasons.compactMap({ $0 }).joined(separator: ", "))")
-//            }
-//
-//            return
-//        }
-//
-//        if provider.isConnected {
-//            provider.disconnect()
-//        }
 
-//        logger?.log("Connecting...")
-//        logger?.log(provider.request)
-//        connectionStateAtomic.set(.connecting)
-    shouldReconnect = true
-
-    DispatchQueue.main.async(execute: engine.connect)
+    startListeningForAppStateUpdates()
   }
 
-  private func reconnect() {
-//        guard connectionState != .reconnecting else {
-//            return
-//        }
-//
-//        connectionStateAtomic.set(.reconnecting)
-//        let maxDelay: TimeInterval = min(0.5 + consecutiveFailures * 2, 25)
-//        let minDelay: TimeInterval = min(max(0.25, (consecutiveFailures - 1) * 2), 25)
-//        consecutiveFailures += 1
-//        let delay = TimeInterval.random(in: minDelay...maxDelay)
-//        logger?.log("⏳ Reconnect in \(delay) sec")
-//
-//        Timer.schedule(timeInterval: delay, queue: provider.callbackQueue) { [weak self] in
-//            self?.connectionStateAtomic.set(.notConnected)
-//            self?.connect()
-//        }
-  }
+  private func startListeningForAppStateUpdates() {
+    NotificationCenter.default.addObserver(
```

Comment:
Why doesn't this use `notificationCenter` variable? Same for line 106 below

Reply:
This has to be always be the default notification center because that's the only one, where these kinds of notifications are published.

Reply:
Hmm, then when will we use `notificationCenter` var?

Reply:
For publishing new events only. We can always use the default one or switch to some custom subscription mechanism if we'll see this is not enough and we need something more custom. That's why I wanted to keep it separate.